internal/main

```diff
@@ -367,17 +380,59 @@ function setup {
 function setup_jenkins {
     echo "Setting up Jenkins build environment"
     echo ""
+    # One-time install of pyenv. It installs to Jenkins home dir, so it can be a shared resource.
+    [ -d "$PYENV_ROOT" ] || {
+        mkdir -p "$PYENV_ROOT"
+        curl -L https://raw.githubusercontent.com/yyuu/pyenv-installer/master/bin/pyenv-installer | bash
+        mkdir -p "$PYENV_ROOT/rax-docs-installed"
+    }
+    export PATH="$PYENV_ROOT/bin:$PATH"
+    eval "$(pyenv init -)" || fail "Installing pyenv seems to have failed. Investigate Jenkins."
+
+    # Python install. We signal to ourselves that an install is done by touching a file. If
+    # the install dir exists but the done file doesn't, we need to wait to see if someone
+    # else is building it at the same time. This is the cost of sharing installations. The
+    # time saved is worth it.
+
+    INSTALL_FILE="$PYENV_ROOT/rax-docs-installed/$PYVERSION"
+    if [ ! -d "$PYENV_ROOT/versions/$PYVERSION" ]; then
+        # Eagerly create the dir to minimize race conditions on this directory test
+        mkdir -p "$PYENV_ROOT/versions/$PYVERSION"
+        SPACE_LEFT=$(df -k --output=avail . | tail -1)
+        [ "$SPACE_LEFT" -gt $((500*1024)) ] || {
+            rmdir "$PYENV_ROOT/versions/$PYVERSION"
+            echo "Less than 500MB remaining in Jenkins home partition."
+            echo "Clean up some space before installing more Pythons."
+            exit 1
+        }
+        # Flags suggested for installing 3.4.3 on CentOS 6 to get past SSL errors
+        export CFLAGS=-I/usr/include/openssl
+        export LDFLAGS=-L/usr/lib64
+        time pyenv install -v "$PYVERSION" && touch "$INSTALL_FILE"
+        du -h --max-depth 1 "$PYENV_ROOT/versions"
+    elif [ -d "$PYENV_ROOT/versions/$PYVERSION" ] && ! [ -f "$INSTALL_FILE" ]; then
+        echo "Python $PYVERSION installation directory exists, but the installation"
+        echo "doesn't seem to have finished. That could mean someone else is"
+        echo "running a build that's building this Python now. To avoid conflicts,"
+        echo "this build will now abort. Try again in a few minutes. If the problem"
+        echo "persists, try deleting $PYENV_ROOT/versions/$PYVERSION and running again."
```

Comment:
Is there any way to delete these versions without logging onto the box?

Reply:
I think the quick fix here will be to `rm -rf "$PYENV_ROOT"` by sticking that in a Jenkinsfile and running it manually. I don't have any other thoughts on dealing with this if/when it happens.